odl/tomo/operators/ray_trafo.py

```diff
@@ -273,299 +158,217 @@ def __init__(self, reco_space, geometry, variant, **kwargs):
-            Default: True
-        kwargs
-            Further keyword arguments passed to the projector backend.
+        # Use impl creation (__cached_impl) when `use_cache` is True
+        if not use_cache or self.__cached_impl is None:
+            # Lazily (re)instantiate the backend
+            self.__cached_impl = self._impl_type(
+                self.geometry,
+                vol_space=self.domain,
+                proj_space=self.range)
 
-        Notes
-        -----
-        The ASTRA backend is faster if data are given with
-        ``dtype='float32'`` and storage order 'C'. Otherwise copies will be
-        needed.
+        return self.__cached_impl
 
-        See Also
-        --------
-        astra_cpu_back_projector
-        AstraCudaBackProjectorImpl
-        skimage_radon_back_projector
-        """
-        domain = kwargs.pop('domain', None)
-        super(RayBackProjection, self).__init__(
-            reco_space=range, proj_space=domain, geometry=geometry,
-            variant='backward', **kwargs)
+    def _call(self, x, out=None, **kwargs):
+        """Forward projection
 
-    def _call_real(self, x_real, out_real, **kwargs):
-        """Real-space back-projection for the current set-up.
+        Parameters
+        ----------
+        x : DiscreteLpElement
+            A volume. Must be an element from `RayTransform.domain`.
+        out : `RayTransform.range` element, optional
+            A DiscreteLpElement from `RayTransform.range`, to which the result
+            of the operator evaluation is written.
+        **kwargs
+            Arbitrary keyword arguments, passed on to the implementation
+            backend.
 
-        This method also sets ``self._astra_backprojector`` for
-        ``impl='astra_cuda'`` and enabled cache.
+        Returns
+        -------
+        DiscreteLpElement
+            A sinogram, from the projection space `RayTransform.range`.
         """
-        if self.impl.startswith('astra'):
-            backend, data_impl = self.impl.split('_')
-            if data_impl == 'cpu':
-                return astra_cpu_back_projector(
-                    x_real, self.geometry, self.range.real_space, out_real,
-                    **kwargs)
-            elif data_impl == 'cuda':
-                if self._astra_wrapper is None:
-                    astra_wrapper = AstraCudaBackProjectorImpl(
-                        self.geometry, self.range.real_space,
-                        self.domain.real_space)
-                    if self.use_cache:
-                        self._astra_wrapper = astra_wrapper
-                else:
-                    astra_wrapper = self._astra_wrapper
 
-                return astra_wrapper.call_backward(x_real, out_real, **kwargs)
-            else:
-                # Should never happen
-                raise RuntimeError('bad `impl` {!r}'.format(self.impl))
+        return self.create_impl(self.use_cache) \
+            .call_forward(x, out, **kwargs)
```

Comment:
```suggestion
        return self.create_impl(self.use_cache).call_forward(
            x, out, **kwargs
        )
```

Reply:
Recently tried to stop with the backslash habit, but with `get_impl` the line fits!